allennlp_models/rc/models/utils.py

```diff
@@ -30,3 +32,16 @@ def get_best_span(span_start_logits: torch.Tensor, span_end_logits: torch.Tensor
     span_start_indices = best_spans // passage_length
     span_end_indices = best_spans % passage_length
     return torch.stack([span_start_indices, span_end_indices], dim=-1)
+
+
+def replace_masked_values_with_big_negative_number(
+    x: torch.Tensor, mask: torch.Tensor, default: float = -1e7
+):
+    """
+    Replace the maskedd values in a tensor something really negative so that they won't
+    affect a max operation.
+    """
+    # We have to be careful here because if we're training with half/mixed precision floats,
+    # -1e7 would result in an overflow. Hence we fall back to the min FP16 value.
+    big_negative_num = max(torch.finfo(x.dtype).min, default)
```

Comment:
We already have [a function for this](https://github.com/allenai/allennlp/blob/32bccfbdaf97045f31861ab16bcfdefb8007c3f2/allennlp/nn/util.py#L1871); why not use it?  I think it's fine to add this new outer function to simplify model logic, I just don't understand this line.

Reply:
I wasn't aware of that function

Reply:
No problem; that's what I was pointing to with my first comment way up above.  Sorry it wasn't clear enough.